texastribune/donations
# Update Salesforce opportunities with gross and net values

#### What's this PR do?

Updates Salesforce to use the gross donation amount as the primary amount in opportunities. Also captures net value.

#### Why are we doing this? How does it help us?

For accounting consistency. For most (all?) other revenue types the gross value is represented in Salesforce opportunities. For Stripe donations it depended on whether the donor chose to pay fees or not whether the gross or net amount was the pprimary "Amount" on the record. Now that "Amount" will always be gross, regardless. And net will also be captured so that fees can be calculated in SF if wanted. This is a request from the Evans.

#### How should this be manually tested?

Make a single test donation and choose not to pay fees. Make another donation of the same amount and choose to pay fees. Look up the Salesforce records. Confirm that the value shown in "Amount" is field is higher for the one where fees were paid by the donor. And confirm that the values for "Net" and "Donor Selected Amount" are correct. 

#### How should this change be communicated to end users?

Make sure that Tommie, Evan Lambert and Anna know. 

#### Are there any smells or added technical debt to note?

No.

#### What are the relevant tickets?

No ticket.

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***

* [x] Added automated tests? *(updated)*

#### TODOs / next steps:

* [ ] After this is merged we'll need to run a job to update all previous donations since 2020-01-01 in SF. This should probably happen shortly after this is merged so reports will be consistent.
